Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/format/xml/XLink.java

```diff
@@ -0,0 +1,576 @@
+     * @param type
+     *            the {@code rdf:type} of the resource
+     * @param title
+     *            a human readable title of the resource
+     * @return the created link
+     */
+    public static Node createResourceLink(Storage storage, Optional<String> localIdentifier, Optional<String> type,
+            Optional<LangString> title) {
+        Node node = storage.createNode();
+        node.put(TYPE, Type.RESOURCE.name);
+        localIdentifier.ifPresent(value -> node.put(LABEL, value));
+        type.ifPresent(value -> node.put(ROLE, value));
+        title.ifPresent(value -> node.put(TITLE, value.getValue()).put(XML.LANG, value.getLanguageTag()));
+        return node;
+    }
+
+    /**
+     * Creates a simple link.
+     *
+     * @param storage
+     *            storage instance to create the link in
+     * @param relation
+     *            Relation description. For example:
+     *            {@code http://purl.org/dc/terms/references}
+     * @param to
+     *            link destination. If the node is a named node, the
+     *            {@code rdf:type}s will be added as {@code xlink:role}s as
+     *            well.
+     * @param title
+     *            link title(s) as human readable text
+     * @param actuate
+     *            when to load the resource
+     * @param show
+     *            context to show the resource in
+     * @return the created link
+     */
+    public static Node createSimpleLink(Storage storage, Optional<IdentifiableNode> relation, IdentifiableNode to,
+            Collection<LangString> title, Optional<Actuate> actuate, Optional<Show> show) {
+
+        return createSimpleLink(storage, relation, to.getIdentifier(),
+            to instanceof Node ? ((Node) to).get(RDF.TYPE).leaves() : Collections.emptySet(), title, actuate, show);
+    }
+
+    /**
+     * Creates a simple link.
+     *
+     * @param storage
+     *            storage instance to create the link in
+     * @param relation
+     *            Relation description. For example:
+     *            {@code http://purl.org/dc/terms/references}
+     * @param to
+     *            link destination
+     * @param type
+     *            Collection of {@code rdf:type}s of the linked resource
+     * @param title
+     *            link title(s) as human readable text
+     * @param actuate
+     *            when to load the resource
+     * @param show
+     *            context to show the resource in
+     * @return the created link
+     */
+    public static Node createSimpleLink(Storage storage, Optional<IdentifiableNode> relation, String to,
+            Collection<String> type, Collection<LangString> title, Optional<Actuate> actuate, Optional<Show> show) {
+        Node node = storage.createNode();
+        node.put(HREF, to);
+        type.forEach(entry -> node.put(ROLE, entry));
+        relation.ifPresent(value -> node.put(ARCROLE, value.getIdentifier()));
+        if (title.size() == 1) {
```

Comment:
Why every of those enums have repeated code? It is repeated in this file 3 times.

Reply:
The keyword `enum` implies that the resulting class is an *immediate* subclass of `java.lang.Enum`. So, sadly, I cannot make an intermediate generic class which summarizes this code. I wish I could.